akka-docs/rst/java/code/docs/http/javadsl/server/directives/BasicDirectivesExamplesTest.java

```diff
@@ -0,0 +1,788 @@
+    //#mapRouteResult
+  }
+
+  public static final class MyCustomRejection implements akka.http.scaladsl.server.Rejection {}
+
+  @Test
+  public void testMapRouteResultPF() {
+    //#mapRouteResultPF
+    final Route route = mapRouteResultPF(
+      new PFBuilder<RouteResult, RouteResult>()
+        .match(Rejected.class, rejected -> {
+          final java.util.List<Rejection> rejections = Collections.singletonList(Rejections.authorizationFailed());
+          return new akka.http.scaladsl.server.RouteResult.Rejected(Util.convertIterable(rejections));
+        }).build(), () -> reject(new MyCustomRejection()));
+
+    // tests:
+    runRouteUnSealed(route, HttpRequest.GET("/"))
+      .assertRejections(Rejections.authorizationFailed());
+    //#mapRouteResultPF
+  }
+
+  @Test
+  public void testMapRouteResultWithPF() {
+    //#mapRouteResultWithPF
+    final Route route = mapRouteResultWithPF(
+      new PFBuilder<RouteResult, CompletionStage<RouteResult>>()
+      .match(Rejected.class, rejected -> CompletableFuture.supplyAsync(() -> {
+        final java.util.List<Rejection> rejections = Collections.singletonList(Rejections.authorizationFailed());
+        return new akka.http.scaladsl.server.RouteResult.Rejected(Util.convertIterable(rejections));
+      })
+    ).build(), () -> reject(new MyCustomRejection()));
+
+    // tests:
+    runRouteUnSealed(route, HttpRequest.GET("/"))
+      .assertRejections(Rejections.authorizationFailed());
+    //#mapRouteResultWithPF
+  }
+
+  @Test
+  public void testMapRouteResultWith() {
+    //#mapRouteResultWith
+    final Route route = mapRouteResultWith(rr -> CompletableFuture.supplyAsync(() -> {
+      if (rr instanceof Rejected) {
+        final java.util.List<Rejection> rejections = Collections.singletonList(Rejections.authorizationFailed());
+        return new akka.http.scaladsl.server.RouteResult.Rejected(Util.convertIterable(rejections));
+      } else {
+        return rr;
+      }
+    }), () -> reject(new MyCustomRejection()));
+
+    // tests:
+    runRouteUnSealed(route, HttpRequest.GET("/"))
+      .assertRejections(Rejections.authorizationFailed());
+    //#mapRouteResultWith
+  }
+
+  @Test
+  public void testPass() {
+    //#pass
+    final Route route = pass(() -> complete("abc"));
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/"))
+      .assertEntity("abc");
+    //#pass
+  }
+
+  private Route providePrefixedStringRoute(String value) {
+    return provide("prefix:" + value, v -> complete(v));
+  }
+
+  @Test
+  public void testProvide() {
+    //#provide
+    final Route route = providePrefixedStringRoute("test");
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/"))
+      .assertEntity("prefix:test");
+    //#provide
+  }
+
+  private Predicate<Rejection> isMethodRejection() {
+    return p -> {
+      if (p instanceof MethodRejection) {
+        return true;
+      } else {
+        return false;
+      }
+    };
+  }
+
+  @Ignore("Test failed")
```

Comment:
This looks right, and the directive looks right. We need to look into this one and figure out why it does not work.